### ISSUE TITLE
wormhole code typo correction via eg soundex

### ISSUE THREAD

**Issue by joeyh** [warner/magic-wormhole#106]
The wormhole receive tab completion is great, but some users may not know about tab completion. And some of us have a hard time spelling words like "phoenetic" :) Wonder if it would be useful to do some kind of typo correction on the entered wormhole code?

Soundex could be used to guess that my "phoenetic" is supposed to be "phonetic" and convert the typoed code into a probably correct one. Of course, it only gets one guess.

The --code option complicates matters, since perhaps the code intentionally mispelled "phoenetic". I don't know what to do about that!

**Comment by warner** [comment]
Yeah, the rule is that people can use whatever codes they want, so the tab-completion is just a hint. I have plans (#26) to let the receiver know what wordlist was used (mostly for i18n), to modify or disable tab-completion when it wouldn't be helpful, so at least it won't drag them in the wrong direction.

I'm intrigued by the soundex idea, though. Do you happen to know if libreadline lets you remove previously-typed characters? I'm imagining something vaguely like how Emacs will tab-complete `foobar<TAB>` into `foolish-bar.txt` if it can find a target filename that can be reached from what you've already typed by adding interstitial characters. For us, it'd be `4-foobar<TAB>` being replaced by `4-foobearish-` if `foobar` was not a prefix of anything, but it was a close soundex match for something else.

Or maybe just modify the readline completion helper function to print extra messages if it sees a close match? So TAB wouldn't automatically change anything, but instead of a list of potential prefix-matching words, you see a `Um, did you mean 'foo'`, like what happens if you run `git blarg` (`git: 'blarg' is not a git command. Did you mean this? "blame"`).

**Comment by joeyh** [comment]
If #26 lets the receive side know what worldlist is being used, it could use soundex automatically to try to correct any words that are not in that wordlist. So the UI would not need to give any feedback.
